### PR TITLE
Introduce Poke Plugins

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

We have been expanding our GTM team, but they frequently struggle with independently debugging issues. This PR introduces a new concept called Plugins in Poke. The objective is straightforward: we want a modular system for adding plugins to our backoffice tool very easily. A plugin can perform tasks ranging from inviting a user to a workspace to stopping a connector or deleting a data source.

In simple terms, we are introducing three endpoints:
1. The first endpoint lists the plugins available for a specific resource type (such as workspace or data_source).
2. The second endpoint retrieves the manifest associated with a plugin, which provides the UI with the schema necessary to display the form required to execute the plugin.
3. The third endpoint runs the plugin using the provided arguments.

All these endpoints strictly ensure that only Dust superusers have the ability to execute this code. In the future, we could envision to implement proper auditing for these plugins and store the records in a database if necessary.

As an example, this PR includes a rewrite of the "Invite user" plugin.
ExamplePlugin

### CODE REVIEW

File: front/components/poke/plugins/PluginForm.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
+import { createIoTsCodecFromArgs } from "@dust-tt/types";
+import { ioTsResolver } from "@hookform/resolvers/io-ts";
+import type * as t from "io-ts";
```
Comment (fontanierh) [inline front/components/poke/plugins/PluginForm.tsx:3]: so no zod ?

File: front/lib/api/poke/plugins/workspaces/invite_user.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { Err, MEMBERSHIP_ROLE_TYPES, Ok } from "@dust-tt/types";
+
+import { handleMembershipInvitations } from "@app/lib/api/invitation";
+import { createPlugin } from "@app/lib/api/poke/types";
+
+export const sendWelcomeEmail = createPlugin(
```
Comment (fontanierh) [inline front/lib/api/poke/plugins/workspaces/invite_user.ts:6]: Very cool

**Review by fontanierh — APPROVED** [review]
Looks very reasonable

File: front/components/poke/plugins/PluginForm.tsx (hunk 1 above)
Comment (flvndvd) [reply front/components/poke/plugins/PluginForm.tsx:3]: Nope not in this PR.

**Pull request merged**